// https://a.yandex-team.ru/api/tree/blob/trunk/arcadia/infra/yp_service_discovery/api/api.proto?repo=arc&rev=4942943

syntax = "proto3";

package sdstub;

message TEndpoint {
    string id = 1;
    string protocol = 2;
    string fqdn = 3;
    string ip4_address = 4;
    string ip6_address = 5;
    int32 port = 6;
    repeated string label_selector_results = 7;
    bool ready = 8;
}

message TEndpointSet {
    string endpoint_set_id = 1;
    repeated TEndpoint endpoints = 2;
}

enum EResolveStatus {
    NOT_EXISTS = 0;
    NOT_CHANGED = 1;
    OK = 2;
    EMPTY = 3;
}

message TReqResolveEndpoints {
    string cluster_name = 1;
    string endpoint_set_id = 2;
    string client_name = 3;
    repeated string label_selectors = 4;
}

message TRspResolveEndpoints {
    uint64 timestamp = 1;
    TEndpointSet endpoint_set = 2;
    EResolveStatus resolve_status = 3;
}

service TServiceDiscoveryService {
    rpc ResolveEndpoints (TReqResolveEndpoints) returns (TRspResolveEndpoints);
}